modules/core/test/src/smithy4s/http/MatchPathSpec.scala

```diff
@@ -72,49 +72,60 @@ class MatchPathSpec() extends munit.FunSuite with munit.ScalaCheckSuite {
     )
     forAll(gen) { case (prefix, segments) =>
       val fullPath = prefix ::: segments.toList
-      val actual = prefix.map(renderExampleSegment).mkString("/")
-      expect.eql(doMatch(fullPath)(actual), None)
+      val actual = prefix.map(renderExampleSegment)
+      expect.eql(doMatch(fullPath)(actual: _*), None)
     }
   }
 
-  test("Is lenient with regards to trailing slashes") {
```

Comment:
Trailing/leading doesn't exist in this area since it's all handled in the uri -> vector conversion